### DISCUSSION TITLE
Unplugin port

### DISCUSSION THREAD

**Discussion by carloitaben** [forge42dev/vite-plugin-icons-spritesheet#18] [Ideas]
Hi there!

First off, kudos for the awesome package 🙂 I've tried out tons of Vite plugins for making sprites and this one nails everything I've been looking for 🫶

I was exploring the code and it seems like it could be ported to [Unplugin](https://unplugin.unjs.io/), which would allow it to be used with other bundlers as well. Next.js has given me Stockholm Syndrome so it would be amazing to use this plugin with Webpack too. Would you be open to the idea of moving it to Unplugin? If that's not something you're interested in I'd be happy to maintain a fork of the project 🙂

I actually gave it a quick shot [here](https://github.com/carloitaben/unplugin-icons-spritesheet). It still needs some more work but the port was pretty straightforward and it’s already functional!

Thanks for considering it, and keep up the great work 🙌

**Comment by AlemTuzlak** [comment]
Thank you so much! 

To be honest I haven't ever looked into Unplugin before as I never worked deeply with Next.js and I despise Webpack but this would be a great way to bring this to everyone on the other hand. 

You said that it still needs some work, what work do you think it needs? I'd be happy to talk about this and investigate a possibility of porting it over.

**Reply by carloitaben** [reply]
Haha, I totally get it, life after Vite is just different. I’d never write a Webpack plugin if it’s not with Unplugin!

As for the work that still needs to be done, there are two main things:
- The `handleHotUpdate` Vite hook needs to be configured according to the [bundler-specific logic guide](https://unplugin.unjs.io/guide/#bundler-specific-logic) since it's a Vite-only hook. I was a bit pressed for time, so I skipped over that part
- There's a type error when consuming the plugin in the test apps. I’m sure there’s a way to fix it, but I just didn’t have time to dig into it

Other than that, the process was pretty straightforward! Here’s what I did:
- I created an [empty Unplugin package](https://unplugin.unjs.io/guide/#creating-an-unplugin-package) to understand their file structure and the `package.json` setup
- I followed their [Nested Plugins guide](https://unplugin.unjs.io/guide/#nested-plugins) to support the multiple configs feature of the plugin
- I moved the `buildStart` and `watchChange` Vite plugin hooks over to the equivalent ones in Unplugin
- I also added a Nextjs site to the `test-apps` directory to try it with Webpack

**Reply by AlemTuzlak** [reply]
I'm definitely open to it but I can't say when I'll have the time to deeply sit down and check it out, I'll try to do it ASAP but no promises. Your fork can work in the meantime, also if you want you can create a PR and we can publish it via pkg.pr.new so people can install it until I have time to look at it.
